In [1]:
import pandas as pd
import numpy as np

from plotly.offline import init_notebook_mode, iplot
import plotly
import plotly.graph_objs as go

init_notebook_mode(connected=True)

In [2]:
data=pd.read_csv(r'G:\__ПРЕЗЕНТАЦИИ\descriptive_stats\clients.csv')
data.head()

ID  Building  Year of sale  Month of sale Type of property  \
0  1030         1          2005             11        Apartment   
1  1029         1          2005             10        Apartment   
2  2002         2          2007              7        Apartment   
3  2031         2          2007             12        Apartment   
4  1049         1          2004             11        Apartment   

   Property number  Area (ft.)           Price  Status Customer ID  ...  \
0               30      743.09   $ 246,172.68    Sold        C0028  ...   
1               29      756.21   $ 246,331.90    Sold        C0027  ...   
2                2      587.28   $ 209,280.91    Sold        C0112  ...   
3               31     1604.75   $ 452,667.01    Sold        C0160  ...   
4               49     1375.45   $ 467,083.31    Sold        C0014  ...   

        Y     M     D  Gender Country       State     Purpose  \
0  1986.0   6.0  21.0       F     USA  California        Home   
1  1983.0   2.0  24.0       F     USA  California        Home   
2  1985.0  12.0  27.0       M     USA  California        Home   
3  1985.0  12.0  27.0       M     USA  California  Investment   
4  1979.0   5.0  15.0       F     USA  California        Home   

   Deal satisfaction Mortgage   Source  
0                  5       No  Website  
1                  5       No  Website  
2                  1      Yes   Client  
3                  3      Yes  Website  
4                  4       No   Agency  

[5 rows x 25 columns]

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 195 entries, 0 to 194
Data columns (total 25 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   ID                       195 non-null    int64  
 1   Building                 195 non-null    int64  
 2   Year of sale             195 non-null    int64  
 3   Month of sale            195 non-null    int64  
 4   Type of property         195 non-null    object 
 5   Property number          195 non-null    int64  
 6   Area (ft.)               195 non-null    float64
 7   Price                    195 non-null    object 
 8   Status                   195 non-null    object 
 9   Customer ID              195 non-null    object 
 10  Entity                   195 non-null    object 
 11  Name                     195 non-null    object 
 12  Surname                  195 non-null    object 
 13  Age at time of purchase  178 non-null    float64
 14  Age Interval             1

In [4]:
# Так как нас интересует портрет человека, уберем данные по компаниям

data = data.drop(data[data.Entity == 'Firm'].index)
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 178 entries, 0 to 177
Data columns (total 25 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   ID                       178 non-null    int64  
 1   Building                 178 non-null    int64  
 2   Year of sale             178 non-null    int64  
 3   Month of sale            178 non-null    int64  
 4   Type of property         178 non-null    object 
 5   Property number          178 non-null    int64  
 6   Area (ft.)               178 non-null    float64
 7   Price                    178 non-null    object 
 8   Status                   178 non-null    object 
 9   Customer ID              178 non-null    object 
 10  Entity                   178 non-null    object 
 11  Name                     178 non-null    object 
 12  Surname                  178 non-null    object 
 13  Age at time of purchase  178 non-null    float64
 14  Age Interval             1

In [5]:
# Убираем знак $ и запятые из цены

def clearPrice(row):
    row['Price'] = row['Price'][3:]
    row['Price'] = row['Price'].replace(',', '')
    return row['Price']
    
data['Price'] = data.apply( clearPrice, axis = 1 )
data.head()

ID  Building  Year of sale  Month of sale Type of property  \
0  1030         1          2005             11        Apartment   
1  1029         1          2005             10        Apartment   
2  2002         2          2007              7        Apartment   
3  2031         2          2007             12        Apartment   
4  1049         1          2004             11        Apartment   

   Property number  Area (ft.)       Price  Status Customer ID  ...       Y  \
0               30      743.09  246172.68    Sold        C0028  ...  1986.0   
1               29      756.21  246331.90    Sold        C0027  ...  1983.0   
2                2      587.28  209280.91    Sold        C0112  ...  1985.0   
3               31     1604.75  452667.01    Sold        C0160  ...  1985.0   
4               49     1375.45  467083.31    Sold        C0014  ...  1979.0   

      M     D  Gender Country       State     Purpose  Deal satisfaction  \
0   6.0  21.0       F     USA  California        Home                  5   
1   2.0  24.0       F     USA  California        Home                  5   
2  12.0  27.0       M     USA  California        Home                  1   
3  12.0  27.0       M     USA  California  Investment                  3   
4   5.0  15.0       F     USA  California        Home                  4   

  Mortgage   Source  
0       No  Website  
1       No  Website  
2      Yes   Client  
3      Yes  Website  
4       No   Agency  

[5 rows x 25 columns]

In [6]:
# Поменяем типы данных цены и возраста

data['Price'] = data.Price.astype('float64')
data['Age at time of purchase'] = data['Age at time of purchase'].astype('int64')
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 178 entries, 0 to 177
Data columns (total 25 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   ID                       178 non-null    int64  
 1   Building                 178 non-null    int64  
 2   Year of sale             178 non-null    int64  
 3   Month of sale            178 non-null    int64  
 4   Type of property         178 non-null    object 
 5   Property number          178 non-null    int64  
 6   Area (ft.)               178 non-null    float64
 7   Price                    178 non-null    float64
 8   Status                   178 non-null    object 
 9   Customer ID              178 non-null    object 
 10  Entity                   178 non-null    object 
 11  Name                     178 non-null    object 
 12  Surname                  178 non-null    object 
 13  Age at time of purchase  178 non-null    int64  
 14  Age Interval             1

In [7]:
# Переведем площадь в м2

data['Area (m)'] = (data['Area (ft.)'].apply(lambda x: x/10.764))
data.head()

ID  Building  Year of sale  Month of sale Type of property  \
0  1030         1          2005             11        Apartment   
1  1029         1          2005             10        Apartment   
2  2002         2          2007              7        Apartment   
3  2031         2          2007             12        Apartment   
4  1049         1          2004             11        Apartment   

   Property number  Area (ft.)      Price  Status Customer ID  ...     M  \
0               30      743.09  246172.68   Sold        C0028  ...   6.0   
1               29      756.21  246331.90   Sold        C0027  ...   2.0   
2                2      587.28  209280.91   Sold        C0112  ...  12.0   
3               31     1604.75  452667.01   Sold        C0160  ...  12.0   
4               49     1375.45  467083.31   Sold        C0014  ...   5.0   

      D Gender  Country       State     Purpose  Deal satisfaction  Mortgage  \
0  21.0      F      USA  California        Home                  5        No   
1  24.0      F      USA  California        Home                  5        No   
2  27.0      M      USA  California        Home                  1       Yes   
3  27.0      M      USA  California  Investment                  3       Yes   
4  15.0      F      USA  California        Home                  4        No   

    Source    Area (m)  
0  Website   69.034745  
1  Website   70.253623  
2   Client   54.559643  
3  Website  149.084913  
4   Agency  127.782423  

[5 rows x 26 columns]

In [15]:
# Найдем типичный пол

data['Gender'].value_counts()

M    108
F     70
Name: Gender, dtype: int64

In [8]:
# Найдем типичную возрастную группу
df_ages = data.groupby('Age Interval ').aggregate({'Age Interval ': "count"})
df_ages.columns = ['Amount']
df_ages

Amount
Age Interval         
18-25               5
26-35              36
36-45              52
46-55              41
56-65              26
65+                18

In [23]:
trace0 = go.Scatter(
    x=df_ages.index,
    y=df_ages.Amount,
    name='Age Interval ')

data = [trace0]
layout = {'title': 'Statistics of age at purchase'}

fig = go.Figure(data=data, layout=layout)

iplot(fig, show_link=False)

In [16]:
# Найдем типичный тип недвижимости

data['Type of property'].value_counts()

Apartment    176
Office         2
Name: Type of property, dtype: int64

In [17]:
# Найдем типичную цель покупки

data['Purpose'].value_counts()

Home          118
Investment     60
Name: Purpose, dtype: int64

In [13]:
# Найдем типичную площадь недвижимости

data['Area (m)'].median()

73.62783351913787

In [14]:
# Найдем типичный чек

data['Price'].median()

245196.73

In [18]:
# Найдем типичный источник лида

df_source = data.groupby('Source').aggregate({'Source': "count"})
df_source.columns = ['Amount']
df_source

Amount
Source         
Agency       58
Client       17
Website     103

In [19]:
labels = df_source.index
values = df_source.Amount

fig = go.Figure(data=[go.Pie(labels=labels, values=values)])
fig.show()

Таким образом можем описать типичного покупателя и покупку.

Пол: мужской
Возрастная группа: 36-45 лет
Тип приобретаемой недвижимости: квартира
Цель покупки: проживание
Площадь недвижимости: 74 кв.м.
Стоимость: 245197 $
Источник лида: вебсайт